cpp/oneapi/dal/algo/knn/test/serialization.cpp

```diff
@@ -129,8 +130,20 @@ class knn_serialization_test : public te::float_algo_fixture<std::tuple_element_
 
     void compare_infer_results(const infer_result<task_t>& actual,
                                const infer_result<task_t>& reference) {
-        INFO("compare responses") {
-            te::check_if_tables_equal<float_t>(actual.get_responses(), reference.get_responses());
+        if constexpr (is_classification) {
+            INFO("compare responses") {
+                te::check_if_tables_equal<float_t>(actual.get_responses(),
+                                                   reference.get_responses());
+            }
+        }
+        if constexpr (is_search) {
+            INFO("compare indices") {
+                te::check_if_tables_equal<float_t>(actual.get_indices(), reference.get_indices());
+            }
+            INFO("compare distances") {
+                te::check_if_tables_equal<float_t>(actual.get_distances(),
+                                                   reference.get_distances());
+            }
         }
 
         // TODO: add "compare distances" checker also
```

Comment:
My understanding is that this TODO fixed now

Reply:
indeed

Reply:
Fixed